src/context.rs

```diff
@@ -258,11 +259,36 @@ impl<'a> Context<'a> {
     }
 
     /// Will lazily get repo root and branch when a module requests it.
-    pub fn get_repo(&self) -> Result<&Repo, git_repository::discover::Error> {
+    pub fn get_repo(&self) -> Result<&Repo, git::discover::Error> {
         self.repo
-            .get_or_try_init(|| -> Result<Repo, git_repository::discover::Error> {
-                let shared_repo =
-                    ThreadSafeRepository::discover_with_environment_overrides(&self.current_dir)?;
+            .get_or_try_init(|| -> Result<Repo, git::discover::Error> {
+                // custom open options
+                let git_open_opts = GitOpenOptions::default()
+                    // trust the git repo and avoid additional checks
+                    .with(git_sec::Trust::Full)
```

Comment:
Actually I was wrong about this, and the way it's currently implemented this value is overridden by the trust level that was determined anyway rendering the `with(…)` adjustment ineffective.

For a moment I was about to neuter the implementation to not be so strict and all in the name of saving a permission check which right now isn't proven to save any perceivable time. All this resulted in [this commit](https://github.com/Byron/gitoxide/commit/270242c707bd086b7746ee45b55791587f1484b1) to add more information on the security implications.

For starship this means it will definitely not accidentally cause the execution of a binary controlled by an attacker, now or in the future, which I think is desirable.

That said, once actual performance measurements of the cost of the permission check (probably mostly on windows) are available, it might be worth reviewing this stance. For now I think it's good to start on the safe side.

Reply:
Ok, I've removed the line. Is `git::Permissions::all()` likely to become an issue? I noticed it includes some DoS protections, which might even be desirable in trusted repos.

Reply:
> Is git::Permissions::all() likely to become an issue?

You are right! It's better to go with the secure defaults for the trust map and only override the pieces that need changing. Maybe like so:

```Rust
let mut options = git_sec::trust::Mapping::default();
options.full.permissions.config = …;
options.reduced.permissions.config = …;
```

It's super valuable to do this as it seems to be easy, even for me, to suggest potentially insecure settings. I am definitely still learning.